fluentd/out_syslog.rb

```diff
@@ -67,11 +67,27 @@ def emit(tag, es, chain)
       tag = tag.sub(@remove_tag_prefix, '') if @remove_tag_prefix
       chain.next
       es.each {|time,record|
-        @packet.hostname = hostname
         if @use_record
-          @packet.facility = record['facility'] || @facilty
-          @packet.severity = record['severity'] || @severity
+          @packet.hostname = record['hostname'] || hostname
+          @packet.severity = (record['level'].eql? 'warning')? 'warn' : record['level'] || @severity
+          if @use_record && record.key?('systemd') && (record['systemd']).key?('u') && (record['systemd']['u']).key?('SYSLOG_FACILITY')
+            fval = record['systemd']['u']['SYSLOG_FACILITY'].to_i
+            if (1..23).include?(fval)
```

Comment:
This is more an FYI that a case statement is unlike other languages and can switch on any conditions that evaluate to true.  I would prefer there use in future to multiple if/else blocks

[1] https://www.skorks.com/2009/08/how-a-ruby-case-statement-works-and-what-you-can-do-with-it/